packages/eslint-plugin/src/rules/return-await.ts

```diff
@@ -237,38 +281,63 @@ export default createRule({
 
       if (option === 'never') {
         if (isAwait) {
+          const fix = (fixer: TSESLint.RuleFixer): TSESLint.RuleFix | null =>
+            removeAwait(fixer, node);
           context.report({
             messageId: 'disallowedPromiseAwait',
             node,
-            fix: fixer => removeAwait(fixer, node),
+            ...(useAutoFix
+              ? { fix }
+              : {
+                  suggest: [
+                    {
+                      messageId: 'disallowedPromiseAwaitSuggestion',
+                      fix,
+                    },
+                  ],
+                }),
           });
         }
 
         return;
       }
 
       if (option === 'in-try-catch') {
-        const isInTryCatch = inTry(expression) || inCatch(expression);
-        if (isAwait && !isInTryCatch) {
+        if (isAwait && !affectsErrorHandling) {
+          const fix = (fixer: TSESLint.RuleFixer): TSESLint.RuleFix | null =>
+            removeAwait(fixer, node);
           context.report({
             messageId: 'disallowedPromiseAwait',
             node,
-            fix: fixer => removeAwait(fixer, node),
+            ...(useAutoFix
+              ? { fix }
+              : {
+                  suggest: [
+                    {
+                      messageId: 'disallowedPromiseAwaitSuggestion',
+                      fix,
+                    },
+                  ],
+                }),
           });
-        } else if (!isAwait && isInTryCatch) {
-          if (inCatch(expression) && !hasFinallyBlock(expression)) {
-            return;
```

Comment:
The return statements were just wrong. Directly caused #8663 and played into #8907 as well.

With the `inCatch`/`hasFinallyBlock` etc logic extracted to the `affectsErrorHandling` variable, they now correctly are included in the preceeding branch (enforcing removing the await).